src/rateMailing.go

```diff
@@ -0,0 +1,65 @@
+package src
+
+import (
+	"go_service/src/common"
+	"time"
+)
+
+type SchedulerTimeGateway interface {
+	common.SchedulerReader
+	common.SchedulerWriter
+}
+
+type RateMailer struct {
+	Es common.EmailSender
+	Sr common.SubscriberListReader
+	Sg SchedulerTimeGateway
+	Cr common.CurrencyReader
+}
+
+func (ms RateMailer) Run() {
```

Comment:
You're trying to implement something like a scheduler here. In your place, I would use a ready-made implemented scheduler, for example, the one I use - “github.com/robfig/cron/v3”.

This way, you don't have to manually process the first and nth iteration of the scheduler. And you also don't need to store the last time it was run in the database (you could even remove it).

Fill free to check my implementation: https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-AlwaysSayNo/blob/main/internal/job/send-emails-job.go